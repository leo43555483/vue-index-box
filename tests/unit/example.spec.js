import Vue from 'vue';
import {
  createMount, trigger, scrollIntoViewFn, elementFromPointFn,
} from './utils';

describe('IndexBox.vue', () => {
  it('touch index && scroll into view', () => {
    const onSelect = jest.fn();
    const wrapper = createMount({
      template: `
      <div>
        <IndexBox @select="onSelect">
          <IndexAnchor title="A" index="A"/>
          <IndexAnchor title="B" index="B"/>
          <IndexAnchor title="C" index="C"/>
        </IndexBox>
      </div>
      `,
      methods: {
        onSelect,
      },
    });
    const indexItems = wrapper.findAll('.ve-index__item');

    const itemWrapper = indexItems.at(0);
    const elementFromPoint = elementFromPointFn(indexItems, 0, 0);
    const scrollFn = scrollIntoViewFn();
    trigger(itemWrapper, 'touchstart', 0, 0);
    expect(elementFromPoint).toBeCalledTimes(1);
    expect(scrollFn).toBeCalledTimes(1);
    expect(onSelect).toBeCalledWith('A');
    Vue.nextTick(() => {
      const tipWrapper = wrapper.find('.ve-index__tip');
      expect(tipWrapper.text()).toBe('A');
    });
  });
});
describe('IndexAnchor.vue', () => {
  it('custom anchor', () => {
    const wrapper = createMount({
      name: 'app',
      template: `
        <div>
          <IndexBox >
            <IndexAnchor title="A" index="A"/>
            <IndexAnchor title="B" index="B"/>
            <IndexAnchor title="C" index="C"/>
          </IndexBox>
        </div>
      `,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
