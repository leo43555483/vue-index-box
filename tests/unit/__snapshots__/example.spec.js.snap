// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexAnchor.vue custom anchor 1`] = `
<div>
  <div class="ve-index__wrapper">
    <ul class="ve-index__list">
      <li class="ve-index__tip" style="display: none;"></li>
      <li data-index="A" class="ve-index__item">A</li>
      <li data-index="B" class="ve-index__item">B</li>
      <li data-index="C" class="ve-index__item">C</li>
      <li data-index="D" class="ve-index__item">D</li>
      <li data-index="E" class="ve-index__item">E</li>
      <li data-index="F" class="ve-index__item">F</li>
      <li data-index="G" class="ve-index__item">G</li>
      <li data-index="H" class="ve-index__item">H</li>
      <li data-index="I" class="ve-index__item">I</li>
      <li data-index="J" class="ve-index__item">J</li>
      <li data-index="K" class="ve-index__item">K</li>
      <li data-index="L" class="ve-index__item">L</li>
      <li data-index="M" class="ve-index__item">M</li>
      <li data-index="N" class="ve-index__item">N</li>
      <li data-index="O" class="ve-index__item">O</li>
      <li data-index="P" class="ve-index__item">P</li>
      <li data-index="Q" class="ve-index__item">Q</li>
      <li data-index="R" class="ve-index__item">R</li>
      <li data-index="S" class="ve-index__item">S</li>
      <li data-index="T" class="ve-index__item">T</li>
      <li data-index="U" class="ve-index__item">U</li>
      <li data-index="V" class="ve-index__item">V</li>
      <li data-index="W" class="ve-index__item">W</li>
      <li data-index="X" class="ve-index__item">X</li>
      <li data-index="Y" class="ve-index__item">Y</li>
      <li data-index="Z" class="ve-index__item">Z</li>
    </ul>
    <div class="ve-index__anchor">
      <div class="ve-index__anchor--inner">A</div>
    </div>
    <div class="ve-index__anchor">
      <div class="ve-index__anchor--inner">B</div>
    </div>
    <div class="ve-index__anchor">
      <div class="ve-index__anchor--inner">C</div>
    </div>
  </div>
</div>
`;
